src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/CheckUtil.java

```diff
@@ -87,7 +87,7 @@ public static Set<String> getSimpleNames(Set<Class<?>> checks) {
             }
 
             return name;
-        }).collect(Collectors.toSet());
+        }).collect(Collectors.toCollection(HashSet::new));
```

Comment:
we can try immutbale here

Reply:
This wouldn't work because `AllChecksTest#testAllModulesAreReferencedInConfigFile` will mutate the collection right?

Reply:
ok, I see that tests use a lot returned collection to add/remove elements from it.
let keep it HashSet